### ISSUE TITLE
Failsafe if all nodes in cluster are failing

### ISSUE THREAD

**Issue by tvoinarovskyi** [aio-libs/aiokafka#459]
After #456 I think it makes sense to put some kind of protection in case all nodes are not available. I feel like using bootstrap nodes list is not a good way to go, rather we should just give an option to fail with an exception if no metadata is available for X period of time, or something similar.

**Comment by dkilgore90** [comment]
@tvoinarovskyi any progress on this? Like in #456 I have kafka brokers which advertise only their ips - which are subject to change on occasion. I have a separate service which notifies me when the ips change, so I have tried to approach this by calling consumer.stop() - but it seems that this hangs if something else takes over the broker's previous ip (perhaps a separate issue), thinking that the broker is coming back but not yet ready.
